File: src/main/java/cn/afeibaili/mchat/MChatSystem.java
package cn.afeibaili.mchat;

import cn.afeibaili.mchat.socket.message.MessageManager;
import cn.afeibaili.mchat.socket.message.SocketManager;

/**
 * MChat系统
 *
 * @author AfeiBaili
 * @version 2026/3/3 01:45
 */

public class MChatSystem extends SocketManager {
    MessageManager<?, ?> messageManager;

    public static MChatSystem INSTANCE;

    public static void stop() {
        INSTANCE.close();
    }

    {
        INSTANCE = this;
    }

    void buildManager(MessageManager<?, ?> messageManager) {
        this.messageManager = messageManager;
    }

    public MessageManager<?, ?> getMessageManager() {
        return messageManager;
    }
}

File: core/src/main/kotlin/online/afeibaili/mchat/MChatSystem.kt
package online.afeibaili.mchat

import online.afeibaili.mchat.socket.SocketManager
import online.afeibaili.mchat.socket.message.MessageManager


/**
 * 接口文件，须实现
 *
 *@author AfeiBaili
 *@version 2025/12/17 13:51
 */


class MChatSystem() : SocketManager() {
    private lateinit var message: MessageManager<*, *>

    init {
        INSTANCE = this
    }

    fun buildManager(messageManager: MessageManager<*, *>) {
        message = messageManager
    }

    fun getMessageManager(): MessageManager<*, *> {
        return message
    }

    companion object {
        lateinit var INSTANCE: MChatSystem

        fun close() {
            INSTANCE.close()
        }
    }
}
